Source/WTF/wtf/posix/FileSystemPOSIX.cpp

```diff
@@ -270,15 +270,17 @@ std::pair<String, PlatformFileHandle> openTemporaryFile(StringView prefix, Strin
     // This is OK for now since the code using it is built on macOS only.
     ASSERT_UNUSED(suffix, suffix.isEmpty());
 
-    char buffer[PATH_MAX];
-    if (snprintf(buffer, PATH_MAX, "%s/%sXXXXXX", temporaryFileDirectory(), prefix.utf8().data()) >= PATH_MAX)
-        goto end;
+    const char* tmp = temporaryFileDirectory();
+    const char* pre = prefix.utf8().data();
+    size_t len = strlen(tmp) + 1 + strlen(pre) + 6 + 1;
+    std::unique_ptr<char, decltype(free)*> buffer(static_cast<char*>(malloc(len)), free);
```

Comment:
This can use `MallocSpan` or any number of other better idioms. Should not call `malloc` and `free` directly like this.